test/distributed/pipelining/test_schedule.py

```diff
@@ -928,6 +929,74 @@ def test_invalid_schedule_missing_action(self):
             _validate_schedule(actions, pp_group_size, num_stages, num_microbatches)
 
 
+class ScheduleUtilTests(TestCase):
+    def test_generate_stage_to_rank_mapping(self):
+        stage_to_rank = generate_stage_to_rank_mapping(2, 2)
+        self.assertEqual(
+            stage_to_rank,
+            {
+                0: 0,
+                1: 1,
+            },
+        )
+        stage_to_rank = generate_stage_to_rank_mapping(2, 4)
+        self.assertEqual(stage_to_rank, {0: 0, 1: 1, 2: 0, 3: 1})
+        stage_to_rank = generate_stage_to_rank_mapping(4, 8)
+        self.assertEqual(
+            stage_to_rank, {0: 0, 1: 1, 2: 2, 3: 3, 4: 0, 5: 1, 6: 2, 7: 3}
+        )
+        stage_to_rank = generate_stage_to_rank_mapping(2, 4, style="v")
+        self.assertEqual(
+            stage_to_rank,
+            {
+                0: 0,
+                1: 1,
+                2: 1,
+                3: 0,
+            },
+        )
+        stage_to_rank = generate_stage_to_rank_mapping(4, 12, style="v")
```

Comment:
Do V-style schedules allow double-looping?  I am not sure if even 4,16 is a valid config but 4,12 looks sus to me.  (But i might be wrong)

Reply:
Ok. I guess I am ok with it if you document it